Sunburst Chart

waxlamp opened the pull request:
This adds a sunburst chart to display the "kingdom"/"species" breakdown in a single view.

This is using a patched version of the sunburst-chart package. Care has been taken to match colors between the existing Google Charts instance and this new sunburst chart.

## UPDATE

Added some new commits:
- white label text, without a halo (@aashish24, please take a look)
- render the chart when data is first selected
- render the chart with the correct dimensions
- debounce chart update when data is changing (avoids rendering problem with inconsistent visual state)

![sunburst2](https://user-images.githubusercontent.com/2903332/60594342-b9c46800-9d72-11e9-8718-825f9effe4fe.png)

waxlamp committed:
Add basic sunburst chart

waxlamp committed:
Render a sunburst chart

waxlamp committed:
Use the same colormap between phylogenetic distribution and sunburst

waxlamp committed:
Make the top-level ring invisible

waxlamp committed:
Use fixed/improved version of sunburst-chart

This is a forked version of the official package (at
@arclamp/sunburst-chart); a PR at the original GitHub repo can be found
here: https://github.com/vasturiano/sunburst-chart/pull/23

matthewma7 commented:
Hi @waxlamp, there are some conflicts, if you don't mind, would you rebase to master? 
After rebasing, you could try it out with the new wider and narrower space.

aashish24 commented:
@waxlamp could also post any screenshot here?

matthewma7 commented:
I am pretty familiar with at creating GIF, I can create one if anyone would like.

waxlamp commented:
@aashish24, I added a screencap to the description. But I encourage you to build the branch and try it out--the interactivity is pretty neat :cat:

waxlamp commented:
@matthewma7, I have rebased and force-pushed the `sunburst` branch; please take a look.

aashish24 commented:
@waxlamp it is super cool looking, cannot wait to try it out. @matthewma7 could you create a short video with sunburst chart that we could send to Neslihan?

matthewma7 commented:
Hi @waxlamp, sometimes when unchecking items the chart would miss some pieces until refresh.
Also, it looks like the fixed size doesn't work so well, it will push below content out of screen and overflow.  could the chart take the container size?

@aashish24 Would we like to fix these issues and, hide the old chart before creating the video?

![2019-07-03_11-24-01](https://user-images.githubusercontent.com/3123478/60604335-76bfc000-9d85-11e9-99d0-0d3d71e2a878.gif)

waxlamp committed:
Render sunburst labels with white text and no contour

waxlamp committed:
Wait until data is selected to render the chart

This also causes the sunburst chart to render with the correct container
size (due to the use of $nextTick() in the mounted() callback).

waxlamp committed:
Debounce sunburst chart update

This seems to avoid the race condition leading to an inconsistent
rendering state when sending too many data updates in too short a time.

matthewma7 commented:
Looks good. 
but I feel the debounce only alleviated the issue because I have seen the issue again when I tried it out, but other than that looks great.
